### ISSUE TITLE
Chef guard updates the config repo for failed cookbooks

### ISSUE THREAD

**Issue by Tech356** [schubergphilis/chef-guard#68]
When I tried to upload a cookbook without bumping the version number, Chef Guard correctly gave me an error and did not upload the cookbook. But it did update the `config` repo. Chef Guard should not update the `config` repo when the cookbook fails to upload. 

```
d:\chef-repo>knife cookbook upload mycookbook
Uploading mycookbook  [0.2.2]
ERROR: Version 0.2.2 of cookbook mycookbook is frozen. Use --force to override.
WARNING: Not updating version constraints for mycookbook in the environment as the cookbook is frozen.
ERROR: Failed to upload 1 cookbook.
```

Here is the diff of commit to `config`.

```
 {
-  "forcedupload": true,
-  "frozen": true,
+  "forcedupload": false,
+  "frozen": false,
   "name": "mycookbook",
-  "source": "https://github.company.com/_nodeload/chef-cookbooks/mycookbook/legacy.tar.gz/master",
+  "source": "N/A",
   "version": "0.2.2"
 }
```

**Comment by svanharmelen** [comment]
Again a great finding! Thanks! I'll be sure to free up some time to build a new release this week to fix the currently found issues...

**Comment by sbotman** [comment]
That's indeed a good one. Lets squash this bug together with #67.

**Comment by svanharmelen** [comment]
Fixed by PR #76
